core/src/main/java/org/apache/iceberg/io/FileWriter.java

```diff
@@ -51,8 +51,9 @@ default void write(Iterable<T> rows) {
    * Writes a row to a predefined spec/partition.
    *
    * @param row a data or delete record
+   * @return PathOffset of written row
    */
-  void write(T row);
+  PathOffset write(T row);
```

Comment:
You mean each abstracted FileWriter will get a `PathOffset` back when append a newly row ? That does not make sense for me because not every writer need this PathOffset to do the following thing.

Reply:
I agree that not every writer need `PathOffset`, but it's required for writing delta data.
For the writer that does not need this, just ignore the return.